test-data/unit/check-class-namedtuple.test

```diff
@@ -587,7 +587,7 @@ class XMethBad(NamedTuple):
 class MagicalFields(NamedTuple):
     x: int
     def __slots__(self) -> None: pass  # E: Cannot overwrite NamedTuple attribute "__slots__"
-    def __new__(cls) -> None: pass  # E: Cannot overwrite NamedTuple attribute "__new__"
+    def __new__(cls) -> None: pass  # E: Name '__new__' already defined on line 8
```

Comment:
This error message is not perfect but probably OK. Why don't you want to keep the old error message?

Reply:
I'd prefer the old error message TBH but the interaction between the namedtuple override checks and the method actually existing seemed nontrivial and I didn't think it was worth fighting with.

Reply:
If it is hard, then I think it is OK.

Reply:
I looked into it a bit and it turns out to be very easy to fix as part of another PR I am putting together. I'm going to land this now with the error message regression (to avoid needing to stack this on top of the other PR, which might require more back-and-forth) and then fix the error message in that PR.